### ISSUE TITLE
WAF efficacy score discussion

### ISSUE THREAD

**Issue by whoot** [fastly/wafefficacy#17]
The WAF efficacy score seems to work not as good as expected. 

Imaging having a WAF which does not filter anything. All XSS payloads are not detected and all true negatives are passed, too.
This would result in an efficacy score of 50% as seen below:
```
-------------XSS-------------
True Positives 0
False Negatives 66
True Negatives 2
False Positives 0
Efficacy 50.0%
```
According to the efficacy score one could say "the WAF is 50% effective in providing protection against XSS attacks." when in reality, the WAF does not provide any protection against XSS attacks.

Same applies to a WAF which just blocks everything:
```
-------------XSS-------------
True Positives 12
False Negatives 0
True Negatives 0
False Positives 34
Efficacy 50.0%
```
Again 50% efficacy, while in reality this would be completely useless.

**Comment by dkegel-fastly** [comment]
Balanced efficacy is a better-than-nothing overall number. 

We can print out a breakdown of positive and negative efficacy so people can drill down past the overall number, but really the better answer may be to include lots more test vectors so we're nowhere near zero on any particular number.

**Comment by whoot** [comment]
I totally understand that you have to use a lot of test data to achieve good results, but in these two (admittedly rather unrealistic) special cases where the WAF does not filter anything or just blocks everything, the dataset won't matter. The efficacy will always be 50%.
What I am trying to say is, that in such cases there should either be a warning - like: "No WAF in use" or "WAF seems to block everything" or something similar - or the efficacy should be 0%, because for obvious reasons.

**Comment by skhalsa-sigsci** [comment]
I agree with your concern about providing some type of waring. In general 50% is a failing score and would likely indicate that your WAF is misconfigured or you need to take a closer look at the test data. The problem with providing a general statement of "No WAF in use" is that you can still receive a 50% without non-zero values. 

```
-------------XSS-------------
True Positives 4
False Negatives 26
True Negatives 26
False Positives 4
Efficacy 50.0%
------------- WAF Efficacy -------------
50.0%
```

**Comment by whoot** [comment]
Indeed a warning would be good in this case.
